### ISSUE TITLE
Update .NET SDK

### ISSUE THREAD

**Issue by DaleMckeown** [actions/runner-images#10964]
### Tool name

.NET SDK

### Tool license

https://dotnet.microsoft.com/en-us/download/dotnet/9.0

### Add or update?

- [ ] Add
- [X] Update

### Desired version

.NET9

### Brief description of tool

.NET SDK needs updating to include the recently released .NET 9 framework.

### URL for tool's homepage

https://dotnet.microsoft.com/en-us/download/dotnet/9.0

### Platforms where you need the tool

- [X] Azure DevOps
- [x] GitHub Actions

### Runner images where you need the tool

- [X] Ubuntu 20.04
- [X] Ubuntu 22.04
- [X] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] macOS 15
- [ ] macOS 15 Arm64
- [X] Windows Server 2019
- [X] Windows Server 2022

**Comment by vidyasagarnimmagaddi** [comment]
Hi @DaleMckeown  , .Net version 9 will be added in next image roll out. thanks

**Comment by DaleMckeown** [comment]
@thomasbach-dk My understanding is they have some specific rules for macOS images, IIRC only 1 version of .NET is allowed on them. It will likely be a decision for MSFT to make themselves when they want to swap that out, so I don't want it to hold up the rest of the images.

Feel free to submit a separate PR targeting macOS specifically.

**Comment by gkizior** [comment]
> Hi @DaleMckeown , .Net version 9 will be added in next image roll out. thanks

ETA?

**Comment by artiomchi** [comment]
@DaleMckeown wait what? I never heard of something like that. Even the existing macos runners have multiple .NET versions installed. 

e.g. https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md#language-and-runtime

**Comment by michal-zatloukal** [comment]
Any update on this? Quite sad to see anyone using GitHub runners in CI not being able to use already released .NET 9 without having to install it every time.

**Comment by DaleMckeown** [comment]
@vidyasagarnimmagaddi When will the new image be released? People want to start using .NET 9.

**Comment by sdcb** [comment]
actually Windows runner is using .NET 9 while others is using .NET 8

**Comment by michaelmairegger** [comment]
I don't get it why every year with each new .NET version we have to use the workaround of manually installing the .NET in the existing images. Why is Microsoft not able to release new runner-images on release date? I think every developer hopes that on day-0 the whole ecosystem is .NET9 ready.

**Comment by DaleMckeown** [comment]
@michaelmairegger It'd be nice if the images were ready for day-0. .NET 9 was release nearly 2 weeks ago now. You'd imagine many people are using GitHub actions/ DevOps pipelines to deploy their applications.

I guess MSFT is a huge organisation (over 230,000 employees apparently) and the thinking is not as joined up as it could/should be.

**Comment by a1-jesper** [comment]
**When** can we expect a resolution?
Seems like there is one ready, but it only needs one more Reviewer Approval to be merged and available.

**Comment by ionixjunior** [comment]
Guys, it's very simple to solve: just add into your pipeline the task "Use .NET Core" and specify the newest version (9.0.101). The specified version will be downloaded and installed in a few seconds. I'm using this in a MAUI app targeting .NET 9 in a macOS 14. It works like a charm!

But I agree with you. It's strange, one month after .NET 9 launch, the SDK is not available yet. I think the same with the .NET workloads for MAUI. They're available only for Windows machines, and every time I'm using macOS to build the app, I need to install them.

**Comment by DoLoop216** [comment]
As @ionixjunior mentions just add step which pins specific version (9.0 in this case).
To achieve this, take a look at [setup-dotnet](https://github.com/actions/setup-dotnet).

TL;DR:
```
      - name: Setup .NET
        uses: actions/setup-dotnet@v3
        with:
          dotnet-version: '9.0.x'
 ```
 
(note: leave `x` as placeholder for patch)

**Comment by slaneyrw** [comment]
> Guys, it's very simple to solve: just add into your pipeline the task "Use .NET Core" and specify the newest version (9.0.101). The specified version will be downloaded and installed in a few seconds. I'm using this in a MAUI app targeting .NET 9 in a macOS 14. It works like a charm!
> 
> But I agree with you. It's strange, one month after .NET 9 launch, the SDK is not available yet. I think the same with the .NET workloads for MAUI. They're available only for Windows machines, and every time I'm using macOS to build the app, I need to install them.

Unfortunately this breaks solutions that multi-target, specifically DotNetCoreCLI task using the 'test' command.

-- Sample Log from build agent.  .NET 8.0 target fails with the same result,  .NET 9.0 passes, but the whole task then fails as some of the result trx files are missing.

Architecture: x64
Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
.NET location: /opt/hostedtoolcache/dotnet/
The following frameworks were found:
  9.0.0 at [/opt/hostedtoolcache/dotnet/shared/Microsoft.NETCore.App]
Learn more:
https://aka.ms/dotnet/app-launch-failed
To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=linux-x64&os=ubuntu.22.04
. Please check the diagnostic logs for more information.
Results File: /home/vsts/work/_temp/_fv-az422-7_2025-01-13_23_25_21.trx

Test Run Aborted.

**Comment by hemanthmanga** [comment]
Hi @DaleMckeown, The latest release of .NET 9 is currently incompatible with Ubuntu and will not be included for the time being. For more details, please refer to the [documentation](https://github.com/actions/runner-images/blob/main/docs/dotnet-ubuntu.md). We are closing this issue, Thank you.

**Comment by DaleMckeown** [comment]
I too would like to know this. I have been manually installing .NET9 on Ubuntu images and have not run into any issues so far.

**Comment by slaneyrw** [comment]
> Hi [@DaleMckeown](https://github.com/DaleMckeown), The latest release of .NET 9 is currently incompatible with Ubuntu and will not be included for the time being. For more details, please refer to the [documentation](https://github.com/actions/runner-images/blob/main/docs/dotnet-ubuntu.md). We are closing this issue, Thank you.

Arbitrarily closing this item with that reason is a real amateur move.  More details needed, given we can install explictly without apparent issues ( multi-targeting aside ).  

The Ubuntu agent is the only Linux host you offer, what else are we supposed to use!

Seriously bad form

**Comment by hemanthmanga** [comment]
> I too would like to know this. I have been manually installing .NET9 on Ubuntu images and have not run into any issues so far.

Hi @DaleMckeown, We have added the .NET 9 version to both Ubuntu 20 and Ubuntu 22. Kindly check the PR. However, for Ubuntu 24, .NET 9 is not yet available in the package manager as it is currently not supported in the built-in Ubuntu feed. If needed, you can install it manually, use [backports](https://launchpad.net/~dotnet/+archive/ubuntu/backports), or utilize setup-dotnet. For more information, please read the complete [documentation](https://github.com/actions/runner-images/blob/main/docs/dotnet-ubuntu.md).

**Comment by michal-zatloukal** [comment]
Why this ticket is closed if .NET 9 is not available in Ubuntu 24? Keep this open until it's done.
